github/admin_users.go

```diff
@@ -13,21 +13,23 @@ import (
 // createUserRequest is a subset of User and is used internally
 // by CreateUser to pass only the known fields for the endpoint.
 type createUserRequest struct {
-	Login *string `json:"login,omitempty"`
-	Email *string `json:"email,omitempty"`
+	Login     *string `json:"login,omitempty"`
+	Email     *string `json:"email,omitempty"`
+	Suspended *bool   `json:"suspended,omitempty"`
 }
 
 // CreateUser creates a new user in GitHub Enterprise.
 //
 // GitHub API docs: https://docs.github.com/enterprise-server@3.11/rest/enterprise-admin/users#create-a-user
 //
 //meta:operation POST /admin/users
-func (s *AdminService) CreateUser(ctx context.Context, login, email string) (*User, *Response, error) {
+func (s *AdminService) CreateUser(ctx context.Context, login, email string, suspended bool) (*User, *Response, error) {
```

Comment:
From the official docs: https://docs.github.com/en/enterprise-server@3.11/rest/enterprise-admin/users?apiVersion=2022-11-28#create-a-user
I'm noticing that both the `email` and `suspended` fields are optional.

Since we are breaking the API with this PR, I'm wondering if it might make more sense to create a new `CreateUserOptions` struct containing these optional fields?  In addition, `login` is required, so maybe change line 16 above to `Login string `json:"login"`.

In other words, we would still use the private `createUserRequest` but we would copy fields in from `opts` where the new signature would be:

```go
unc (s *AdminService) CreateUser(ctx context.Context, login string, opts *CreateUserOptions) (*User, *Response, error) {
```

If we do this now, then we might have a bit more flexibility in the future to add new fields to the `CreateUserOptions` struct without breaking the API again.

Thoughts?

Reply:
Hi Glenn - I like the idea to break it only once if we have to break it at all.

Taking a look at other examples around the codebase, it looks like one pattern is to pass URL path segments individually, then the whole object that would end up in the request body. Examples

* [TeamsService.CreateTeam(ctx context.Context, org string, team NewTeam)](https://github.com/google/go-github/blob/e9f52699f5e55a4791567bce668f56446e6343ef/github/teams.go#L186)
* [PullRequestService.Create(ctx context.Context, owner string, repo string, pull *NewPullRequest)](https://github.com/google/go-github/blob/e9f52699f5e55a4791567bce668f56446e6343ef/github/pulls.go#L268)

Following this pattern, it seems a more consistent option would be to export the `createUserRequest` object and pass that as the second param (after `ctx`). 

WDYT?

Reply:
Ah, yes, I see what you are saying.  Path query parameters should be listed as separate arguments, and then you can have another parameter:

```go
func (s *AdminService) CreateUser(ctx context.Context, req CreateUserRequest) (*User, *Response, error) {
```

where:

```go
// CreateUserRequest represents the fields sent to the `CreateUser` endpoint.
// Note that `Login` is a required field.
type CreateUserRequest struct {
	Login     string `json:"login"`
	Email     *string `json:"email,omitempty"`
	Suspended *bool   `json:"suspended,omitempty"`
}
```